content/articles/2017-12-14-styler-1-0-0.Rmd

```diff
@@ -0,0 +1,237 @@
+* `style_pkg()` styles the source files of an R package.
+* An RStudio Addin that styles the active R or Rmd file, the current 
+  package or the highlighted code.
+
+Note that all functions except `style_text()` modify *in place*, that is, they overwrite the source. `style_file()` for example modifies the files it is given without creating a backup copy. We therefore strongly advice to use version control in conjunction with styler and consider the implications this might have on your workflow.
+
+## styling options
+
+A distinguishing feature of styler is its flexibility. The following options are available:
+
+* `scope`. What is to be styled, e.g. just spacing or also indention, line breaks, tokens?
+* `strict`. How strict should be rules be applied?
+* `math_token_spacing`. Granular control for spacing around math tokens.
+* `reindention`. Allows to set the indention of certain regex patterns manually.
+
+We will briefly describe all of them below.
+
+### scope
+
+`scope` can take one of the following values:
+
+* "none": Performs no transformation at all.
+* "spaces": Manipulates spacing between token on the same line.
+* "indention": In addition to "spaces", this option also manipulates the indention level.
+* "line_breaks": In addition to "indention", this option also manipulates line breaks.
+* "tokens": In addition to "line_breaks", this option also manipulates tokens.
+
+Hence, `scope < "tokens"` will never change code in the sense that the expression remains the same before and after styling, as only spaces and line breaks between the tokens are modified. styler can internally validate the styling, so you can be sure that it has not broken your code. Although extensively tested, we recommend to inspect the styling result with `scope = "tokens"`, since in this case, an automatic validation is not possible.
+
+Let's now see a few examples. We can limit ourselves to styling just spacing information:
+
+```{r}
+library("styler")
+style_text("a=3; 2", scope = "spaces")
+```
+
+Or, on the other extreme of the scale, styling spaces, indention, line breaks 
+and tokens:
+
+```{r}
+style_text("a=3; 2", scope = "tokens")
+```
+
+In the next example, we want to style spaces and indention, but not line breaks
+and tokens:
+```{r}
+style_text(
+  "a=  function( x) {
+  'x'
+  }", 
+  scope = "indention"
+)
+```
+
+
+### strict
+
+Another option that is helpful to determine the level of 'invasiveness' is `strict`. If set to `TRUE`, spaces and line breaks before or after tokens are set to either zero or one. However, in some situations, this might be undesirable (so we set `strict = FALSE`), as the following example shows:
+
+```{r}
+style_text(
+  "data_frame(
+     small  = 2 ,
+     medium = 4,#comment without space
+     large  =6
+   )", 
+  strict = FALSE
+)
+```
+
+We prefer to keep the equal sign after "small", "medium" and large aligned, so we set `strict = FALSE` to set spacing to *at least* one around `=`.
```

Comment:
"large" should be in quotes, to be parallel with "small" and "medium"

Reply:
Thanks for reviewing so carefully, I am really sorry that I waste your time with grammar mistakes of a non-native speaker. Should have reviewed myself more carefully.